controllers/replicationdestination_controller.go

```diff
@@ -232,153 +170,124 @@ func (r *ReplicationDestinationReconciler) SetupWithManager(mgr ctrl.Manager) er
 		Complete(r)
 }
 
-//nolint:dupl
-func updateNextSyncDestination(
-	rd *volsyncv1alpha1.ReplicationDestination,
-	metrics volsyncMetrics,
-	logger logr.Logger,
-) (bool, error) {
-	// if there's a schedule, and no manual trigger is set
-	if rd.Spec.Trigger != nil &&
-		rd.Spec.Trigger.Schedule != nil &&
-		rd.Spec.Trigger.Manual == "" {
-		parser := cron.NewParser(cron.Minute | cron.Hour | cron.Dom | cron.Month | cron.Dow | cron.Descriptor)
-		schedule, err := parser.Parse(*rd.Spec.Trigger.Schedule)
-		if err != nil {
-			logger.Error(err, "error parsing schedule", "cronspec", rd.Spec.Trigger.Schedule)
-			return false, err
-		}
-
-		// If we've previously completed a sync
-		if !rd.Status.LastSyncTime.IsZero() {
-			if pastScheduleDeadline(schedule, rd.Status.LastSyncTime.Time, time.Now()) {
-				metrics.OutOfSync.Set(1)
-			}
-			next := schedule.Next(rd.Status.LastSyncTime.Time)
-			rd.Status.NextSyncTime = &metav1.Time{Time: next}
-		} else { // Never synced before, so we should ASAP
-			rd.Status.NextSyncTime = &metav1.Time{Time: time.Now()}
-		}
-	} else { // No schedule, so there's no "next"
-		rd.Status.NextSyncTime = nil
+func newRDMachine(rd *volsyncv1alpha1.ReplicationDestination, c client.Client,
+	l logr.Logger, er events.EventRecorder) (*rdMachine, error) {
+	dataMover, err := mover.GetDestinationMoverFromCatalog(c, l, er, rd)
+	if err != nil {
+		return nil, err
 	}
 
-	if rd.Status.LastSyncTime.IsZero() {
-		// Never synced before, so we're out of sync
-		metrics.OutOfSync.Set(1)
+	metrics := newVolSyncMetrics(prometheus.Labels{
+		"obj_name":      rd.Name,
+		"obj_namespace": rd.Namespace,
+		"role":          "destination",
+		"method":        dataMover.Name(),
+	})
+
+	if rd.Status == nil {
+		rd.Status = &volsyncv1alpha1.ReplicationDestinationStatus{}
```

Comment:
The status is still initialized in the Reconcile() - should we remove one of them? (same for replicationsource_controller)

Reply:
It's done in Reconcile() so that we can set the Condition to error if necessary.
I had also added it to here because none of the methods check before they access status, but that's a good point that we're guaranteed it won't be nil due to Reconcile. Will remove this one.

Reply:
done